src/runtime-rs/crates/runtimes/virt_container/src/container_manager/manager.rs

```diff
@@ -272,4 +277,54 @@ impl ContainerManager for VirtContainerManager {
         process.process_type == ProcessType::Container
             && process.container_id.container_id == self.sid
     }
+
+    // if guest_swap is true, add swap to memory_sandbox
+    // returns `(memory_sandbox_b, swap_sandbox_b)` (in Byte)
+    async fn mem_required(&self) -> Result<(u64, i64)> {
+        let mut _mem_sandbox = 0;
+        let mut _swap_sandbox = 0;
+
+        // todo(memory hotplug): actual calculation logic
+
+        Ok((_mem_sandbox, _swap_sandbox))
+    }
+
+    // calculates the total required vpus by adding each containers' requirement within the pod
+    async fn cpus_required(&self) -> Result<u32> {
+        let mut total_vcpu = 0;
+        let mut cpuset_count = 0;
+        let cpu_mem_info = self.resource_manager.sandbox_cpu_mem_info().await;
+
+        let containers = self.containers.read().await;
+        for c in containers.values() {
+            if let Some(resource) = &c.linux_resources {
+                if let Some(cpu) = &resource.cpu {
```

Comment:
@Ji-Xinyou @quanweiZhou we should also check the container state here, refer to [go runtime](https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/sandbox.go#L2197). If the container state is stopped, we don't need to count the CPU here. To get the container state, we have to do it in the container manager, so I think it's inappropriate to implement it in Sandbox, wdyt @quanweiZhou ?